docs/protocol/architecture.mdx

```diff
@@ -0,0 +1,169 @@
+  style OnChain fill:#ffe1f5,stroke:#ffbde6,color:#111
+  style Blockchain fill:#f0f0f0,stroke:#c9c9c9,color:#111
+
+  %% Emphasize the RPC link (index 2 in the edge list above)
+  linkStyle 2 stroke:#0ea5e9,stroke-width:2px
+```
+
+## Communication Patterns
+
+### Channel Opening
+
+The <Tooltip content="A secure communication pathway between participants that locks funds in an on-chain smart contract while enabling off-chain state updates.">channel</Tooltip> opening process follows a coordinated sequence between client and <Tooltip content="A virtual ledger layer that provides a unified ledger (through Nitro RPC) and coordinates state channels (through Nitrolite), providing chain abstraction for developers and users.">a clearnode</Tooltip>:
+
+1. Client requests channel creation from <Tooltip content="A virtual ledger layer that provides a unified ledger (through Nitro RPC) and coordinates state channels (through Nitrolite), providing chain abstraction for developers and users.">a clearnode</Tooltip> via <Tooltip content="The off-chain communication protocol.">Nitro RPC</Tooltip>
+2. The clearnode returns signed initial <Tooltip content="A snapshot of the channel at a point in time, including fund allocations and application-specific data.">state</Tooltip> (signature at index 1)
+3. Client signs the initial <Tooltip content="A snapshot of the channel at a point in time, including fund allocations and application-specific data.">state</Tooltip> (signature at index 0)
+4. Client submits transaction to <Tooltip content="The underlying distributed ledger technology (e.g., Ethereum).">blockchain</Tooltip> with both signatures
+5. Contract verifies signatures and emits `Opened` event
+6. Channel becomes ACTIVE immediately
+7. The clearnode monitors the `Opened` event and updates its internal state
+
+```mermaid
+sequenceDiagram
+    participant Client
+    participant Clearnode
+    participant Blockchain
+    
+    Client->>Clearnode: create_channel request
+    Clearnode->>Client: signed initial state (clearnode signature)
+    Client->>Client: Add own signature
+    Client->>Blockchain: create() with BOTH signatures
+    Blockchain->>Blockchain: Verify signatures<br/>Set status = ACTIVE
+    Blockchain->>Blockchain: Emit Opened event
+    Blockchain-->>Clearnode: Opened event (monitored)
+    Clearnode->>Client: channel_update notification
+```
+
+:::success Cooperative Opening
+Channel opening requires cooperation between both parties, ensuring mutual agreement before funds are locked.
+:::
+
+### Off-Chain Updates
+
+**Off-Chain Updates**:
+
+1. <Tooltip content="An entity (identified by a wallet address) that is part of a channel.">Participants</Tooltip> exchange signed <Tooltip content="A snapshot of the channel at a point in time, including fund allocations and application-specific data.">state</Tooltip> updates via <Tooltip content="The off-chain communication protocol.">Nitro RPC</Tooltip>
```

Comment:
I have noticed that a "state" with the same tooltip is used interchangeably across on-chain and off-chain channels. I guess this creates confusion for readers making them think this is the same structure. I think this can be fixed by differentiating those terms as "state channel state" / "on-chain state" and "app session state" / "off-chain state".
@ihsraham @dimast-x @philanton , WDYT?